data_model/src/domain.rs

```diff
@@ -108,6 +126,42 @@ impl Domain {
             logo: None,
         }
     }
+
+    /// Returns a reference to the account corresponding to the account id.
+    pub fn get_account(&self, account_id: &<Account as Identifiable>::Id) -> Option<&Account> {
+        self.accounts.get(account_id)
+    }
+
+    /// Returns a mutable reference to the account corresponding to the account id.
+    pub fn get_account_mut(
+        &mut self,
+        account_id: &<Account as Identifiable>::Id,
+    ) -> Option<&mut Account> {
+        self.accounts.get_mut(account_id)
+    }
+
+    /// Adds account into the domain
+    pub fn add_account(&mut self, account: Account) -> Option<Account> {
+        self.accounts.insert(account.id.clone(), account)
+    }
+
+    /// Removes account from the domain
+    pub fn remove_account(
+        &mut self,
+        account_id: &<Account as Identifiable>::Id,
+    ) -> Option<Account> {
+        self.accounts.remove(account_id)
+    }
+
+    /// Gets an iterator over accounts of the domain
+    pub fn accounts(&self) -> impl Iterator<Item = &Account> {
+        self.accounts.values()
+    }
+
+    /// Gets a mutable iterator over accounts of the domain
+    pub fn accounts_mut(&mut self) -> impl Iterator<Item = &mut Account> {
+        self.accounts.values_mut()
+    }
```

Comment:
I think mutable iterator over accounts is only required for `iroha_core` use cases. I don't think it has much use when writing smart contracts or client applications

Reply:
but this doesn't matter much

Reply:
Only add mutation if we can't get around it. Immutable is fine.